### PR TITLE
Generalize list signatures

### LINKED ISSUE #46: Same signature for all lists ?
Currently, only the list of records can be sorted, filtered, paginated.

Since the bucket, collection, and group list share the same capabilities (it's implemented [using the same resource](http://kinto.readthedocs.org/en/latest/concepts.html#technical-architecture)), we could also apply the same behaviour to every list method :)

(not a priority though).

### PR DESCRIPTION
**Pull request by leplatrem**
Fixes #46

* [x] Generalize code
* [x] Leverage code in list of records
* [x]  Leverage code in list of buckets
* [x]  Leverage code in list of groups
* [x]  Leverage code in list of collections
* [x] Wait https://github.com/Kinto/kinto/pull/718
* [x] Update documentation

### CODE REVIEW

File: test/integration_test.js [hunk 1]
```diff
@@ -220,18 +220,72 @@ describe("Integration tests", function() {
       });
     });
 
-    describe("#listBuckets", () => {
+    describe.only("#listBuckets", () => {
```
Comment (leplatrem) [inline test/integration_test.js]: leftover

File: src/base.js [hunk 2]
```diff
@@ -366,6 +366,70 @@ export default class KintoClientBase {
     return options.raw ? promise : promise.then(({json}) => json);
   }
 
+  paginatedList(path, listOptions, httpOptions) {
```
Comment (leplatrem) [inline src/base.js]: * listOptions -> params
* httpOptions -> options

**Comment by leplatrem** [comment]
@n1k0 final r+ ?

**Comment by n1k0** [comment]
r+ \o/

**Pull request merged**